pyroomacoustics/whitening.py

```diff
@@ -0,0 +1,40 @@
+# @version: 1.0  2018, Juan Azcarreta
+# Whitening algorithm as a preprocessing step for blind source separation
+
+from __future__ import division
+
+import numpy as np
+
+#=========================================================================
+# Whitening in the time-frequency domain.
+#=========================================================================
+
+def whitening(X):
+    '''
+    This function computes the time-frequency domain decorrelation (whitening)
+    Withening based on the method presented in Section II of the following paper:
+    Cichocki, Andrzej & Osowski, S & Siwek, Krzysztof. (2004). Prewhitening Algorithms of Signals
+    in the Presence of White Noise.
+
+    Parameters
+    ----------
+    X: ndarray (nframes, nfrequencies, nchannels)
+        STFT representation of the observed signal
+        n_src: int, optional
+        The number of sources or independent components
+    Returns
+    ----------
+    Y: ndarray (nframes, nfrequencies, nchannels)
+        STFT representation of the whitened signal
+    -------
+    '''
+
+    [n_frames, n_freq, n_chan] = X.shape
+    Y = np.zeros([n_frames, n_freq, n_chan], dtype=np.complex)
+    fudge = 1E-16

```

Comment:
haha didn't know this term `fudge`! Maybe `tol` is more appropriate though. I think we've use that terminology in other part of the package.